src/pages/index.js

```diff
@@ -4,27 +4,113 @@ import "../pages/index.css";
 import FormValidator from "../components/FormValidator.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupConfirmDelete from "../components/ConfirmDelete.js";
+import PopupWithProfileImage from "../components/PopupWithProfileImage.js";
+import Api from "../components/Api.js";
 import UserInfo from "../components/UserInfo.js";
 import { initialCards, options } from "../utils/constants";
 
 // Initialize user information
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__job",
+  profileImageSelector: ".profile__image",
 });
 
 // Form elements and validators
 const profileFormElement = document.forms["modal__profile-form"];
 const cardFormElement = document.forms["modal__card-form"];
 const profileFormValidator = new FormValidator(options, profileFormElement);
 const cardFormValidator = new FormValidator(options, cardFormElement);
+
+const deleteCardFormElement = document.forms["modal__delete-card-form"];
+const editProfileFormElement = document.forms["modal__profile-image"];
+const deleteCardFormValidator = new FormValidator(
+  options,
+  deleteCardFormElement
+);
+const editProfileFormValidator = new FormValidator(
+  options,
+  editProfileFormElement
+);
+
 const popupWithImage = new PopupWithImage("#modal_image");
+const userImage = document.querySelector(".profile__image");
+
+// Initialize the profile image modal
+const popupUpdateProfileImage = new PopupWithProfileImage({
+  popupSelector: "#modal_profile-image",
+  imageSelector: ".profile__image-container",
+  handleFormSubmit: (formData) => {
+    return api
+      .updatingProfile({ profileImage: formData.profileImage })
+      .then((updatedUserData) => {
+        userInfo.setUserInfo(updatedUserData);
+      })
+      .catch((err) => {
+        console.error(`Failed to update profile image: ${err}`);
+      });
```

Comment:
* The `avatar` modal doesn't close after submission

It's because you should reuse `PopupWithForm` for creating `popupUpdateProfileImage`


It has everything you need: the text changing, closing, resetting and so on